mqemitter.js

```diff
@@ -130,7 +130,9 @@ MQEmitter.prototype._do = function (message, callback) {
   const matches = this._matcher.match(message.topic)
 
   this.current++
-  this._parallel(this, matches, message, callback)
+  setImmediate(
+    this._parallel.bind(this, this, matches, message, callback)
+  )
```

Comment:
Adding a setImmediate here will definitely cripple the performance of this module. A different solution is needed.

Reply:
Suggestions?

Reply:
Suggestions? process.nextTick ?

Reply:
Something way more convoluted. We should test if this was called in the same tick more than x times and then use setImmediate just in that case.